src/internal/methods/NeoEsp32LcdXMethod.h

```diff
@@ -0,0 +1,551 @@
+            LCD_CAM.lcd_rgb_yuv.lcd_conv_bypass = 0; // Disable RGB/YUV converter
+            LCD_CAM.lcd_misc.lcd_next_frame_en = 0;  // Do NOT auto-frame
+            LCD_CAM.lcd_data_dout_mode.val = 0;      // No data delays
+            LCD_CAM.lcd_user.lcd_always_out_en = 1;  // Enable 'always out' mode
+            LCD_CAM.lcd_user.lcd_8bits_order = 0;    // Do not swap bytes
+            LCD_CAM.lcd_user.lcd_bit_order = 0;      // Do not reverse bit order
+            LCD_CAM.lcd_user.lcd_2byte_en = 0;       // 8-bit data mode
+            LCD_CAM.lcd_user.lcd_dummy = 1;          // Dummy phase(s) @ LCD start
+            LCD_CAM.lcd_user.lcd_dummy_cyclelen = 0; // 1 dummy phase
+            LCD_CAM.lcd_user.lcd_cmd = 0;            // No command at LCD start
+            // Dummy phase(s) MUST be enabled for DMA to trigger reliably.
+
+            // Set up DMA descriptor list (length and data are set before xfer)
+            desc = (dma_descriptor_t *)LcdBuffer; // At start of alloc'd buffer
+            for (int i = 0; i < num_desc; i++) {
+                desc[i].dw0.owner = DMA_DESCRIPTOR_BUFFER_OWNER_DMA;
+                desc[i].dw0.suc_eof = 0;
+                desc[i].next = &desc[i + 1];
+            }
+            desc[num_desc - 1].dw0.suc_eof = 1;
+            desc[num_desc - 1].next = NULL;
+
+            // Alloc DMA channel & connect it to LCD periph
+            gdma_channel_alloc_config_t dma_chan_config = {
+                .sibling_chan = NULL,
+                .direction = GDMA_CHANNEL_DIRECTION_TX,
+                .flags = {.reserve_sibling = 0}};
+            gdma_new_channel(&dma_chan_config, &dma_chan);
+            gdma_connect(dma_chan, GDMA_MAKE_TRIGGER(GDMA_TRIG_PERIPH_LCD, 0));
+            gdma_strategy_config_t strategy_config = {.owner_check = false,
+                                                        .auto_update_desc = false};
+            gdma_apply_strategy(dma_chan, &strategy_config);
+
+            // Enable DMA transfer callback
+            gdma_tx_event_callbacks_t tx_cbs = {.on_trans_eof = dma_callback};
+            gdma_register_tx_event_callbacks(dma_chan, &tx_cbs, NULL);
+        }
+    }
+
+    void Destruct()
+    {
+        if (LcdBuffer == nullptr)
+        {
+            return;
+        }
+
+        // TODO: destroy peripheral pin settings
+        // i2sSetPins(busNumber, -1, -1, -1, false);
+        // i2sDeinit(busNumber);
+
+        gdma_reset(dma_chan);
+
+        heap_caps_free(LcdBuffer);
+
+        LcdBufferSize = 0;
+        LcdBuffer = nullptr;
+
+        MuxMap.Reset();
+    }
+
+    void ResetBuffer()
+    {
+        // to keep the inner loops for EncodeIntoDma smaller
+        // they will just OR in their values
+        // so the buffer must be cleared first
+        if (MuxMap.IsNoMuxBusesUpdate())
+        {
+            // clear all the data in preperation for each mux channel to add
+            memset(LcdBuffer, 0x00, LcdBufferSize);
```

Comment:
This needs to only clear the DMA buffer so it doesn't wipe the descriptors if there is more than 1 write. **